File: from-3101-to-3200/3191-minimum-operations-to-make-binary-array-elements-equal-to-one-i/solution.go
func minOperations(nums []int) int {
    result := 0
    n := len(nums)
    for i := 0; i < n - 2; i ++ {
        if nums[i] == 0 {
            result += 1
            flip(&nums[i + 1])
            flip(&nums[i + 2])
        }
    }
    if nums[n - 2] == 0 || nums[n - 1] == 0 {
        return -1
    }
    return result
}

func flip(a *int) {
    *a = (*a + 1) % 2
}

File: from-3101-to-3200/3191-minimum-operations-to-make-binary-array-elements-equal-to-one-i/solution.py
class Solution:
    def minOperations(self, nums: List[int]) -> int:
        result = 0
        for i in range(len(nums) - 2):
            if nums[i] == 0:
                nums[i + 2] ^= 1
                nums[i + 1] ^= 1
                result += 1

        if nums[-2] == 0 or nums[-1] == 0:
            return -1

        return result